pkg/tfvars/openstack/rhcos_image.go

```diff
@@ -0,0 +1,139 @@
+package openstack
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imagedata"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imageimport"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/sirupsen/logrus"
+)
+
+// uploadBaseImage creates a new image in Glance and uploads the RHCOS image there
+func uploadBaseImage(cloud string, localFilePath string, imageName string, clusterID string) error {
+	logrus.Debugln("Creating a Glance image for RHCOS...")
+
+	f, err := os.Open(localFilePath)
+	if err != nil {
+		return err
+	}
+
+	opts := clientconfig.ClientOpts{
+		Cloud: cloud,
+	}
+
+	conn, err := clientconfig.NewServiceClient("image", &opts)
+	if err != nil {
+		return err
+	}
+
+	imageCreateOpts := images.CreateOpts{
+		Name:            imageName,
+		ContainerFormat: "bare",
+		DiskFormat:      "raw",
```

Comment:
Shouldn't `DiskFormat` depends on the actual image type rather than hardcoding it to raw?

I find it weird that we need to set this when importing an image, because the whole point is to let glance decide what it the best image format for the backend...

Reply:
Yeah, it should be `qcow2` here for sure. The image conversion plugin will change it if necessary upon image activation.